Activities/Cryptography/UiPath.Cryptography.Activities/DecryptFile.cs

```diff
@@ -126,20 +144,39 @@ protected override void Execute(CodeActivityContext context)
                     throw new ArgumentException(Resources.FileAlreadyExistsException, Resources.OutputFilePathDisplayName);
                 }
 
-                byte[] encrypted = File.ReadAllBytes(inputFilePath);
-
-                byte[] decrypted = null;
-                try
+                if (inputFile != null && !inputFile.IsFolder)
                 {
-                    decrypted = CryptographyHelper.DecryptData(Algorithm, encrypted, CryptographyHelper.KeyEncoding(keyEncoding, key, keySecureString));
+                    // Get local file
+                    var localFile = inputFile.ToLocalResource();
+                    Task.Run(async () => { await localFile.ResolveAsync(); });
```

Comment:
This will not wait for the resolve to actually happen. Please either:

a) derive this from an async actvitiy, see how it's done in the Activities repo.
b) do Task.Run(..).GetAwaiter().GetResult() to make sure we actually wait for the resolve to complete. You can use `ResolveAndGetUnderlyingFile` for this or something similar, to make sure we don't needlessly duplicate code.

Same in other places this happens.